### DISCUSSION TITLE
Why do songs force a certain number of keys for a song?

### DISCUSSION THREAD

**Discussion by SuPeRDuDe5898** [ppy/osu#26472] [Q&A]
I just wanted to play mania 4k like usual, until I played "Miracle Infinity", which forced 6 keys even with the 4k mod, is there a reason why that is?

**Comment by cihe13375** [comment]
4k mod only works for converts (i.e. maps that are originally designed for std and you play them in mania mode). It has no effect on mania specific maps.

**Reply by peppy** [reply]
Really the mods should be disabled when they can't be used. Or made to work.
